src/app/entities/courses/components/course-item/course-item.component.ts

```diff
@@ -20,11 +22,25 @@ export class CourseItemComponent {
   deleteCourseIcon: IconDefinition = faTrash;
   topRatedIcon: IconDefinition = faStar;
 
-  constructor() {
+  constructor(private router: Router) {
     this.courseDeleted = new EventEmitter<number>();
   }
 
+  onCourseEditClick(courseId: number, courseTitle: string) {
+    this.navigateToCourseEditPage(courseId, courseTitle);
+  }
+
   onCourseDeleteClick(courseId: number) {
+    this.emitCourseDeletedEvent(courseId);
+  }
+
+  private navigateToCourseEditPage(courseId: number, courseTitle: string) {
+    this.router.navigate([ `${AppRoutePath.Courses}/${courseId}` ], {
+      state: { breadcrumbsStepTitle: courseTitle },
+    });
+  }
+
+  private emitCourseDeletedEvent(courseId: number) {
     this.courseDeleted.emit(courseId);
   }
```

Comment:
it is not necessary as it does not bring any additional value/input/operation.

Reply:
Got it, thanks